### PR TITLE
Support single encoder models

### PR DESCRIPTION
**Pull request by jeremystan**
## What

Currently Keras lore models must have multiple encoders, both because `build_embedding_layer` calls `Concatenate` on embeddings, and because the pipeline encoders are iterated through at multiple points in the code. This change tests and supports declaring a single encoder model both as any of the following:

```
# A tuple
def get_encoders(self):
    return (
        Unique('column'),
    )

# A length 1 tuple
def get_encoders(self):
    return (
        Unique('column')
    )

# A single encoder
def get_encoders(self):
    return Unique('column')
```

Note that the first example (a tuple) works if you only make the change to `build_embedding_layer`, but the changes to `holdout.py` is required to support the 2nd and 3rd examples.

## Why

Sometimes you only have one feature / encoder, especially for language models or baseline models.

@montanalow

### CODE REVIEW

File: lore/pipelines/holdout.py [hunk 1]
```diff
@@ -90,6 +90,12 @@ def encoders(self):
         if self._encoders is None:
             with timer('fit encoders'):
                 self._encoders = self.get_encoders()
+
+                # Ensure we have an iterable for all single encoder cases
+                if self._encoders.__class__ is not tuple:
```
Comment (montanalow) [inline lore/pipelines/holdout.py]: Can we make this a check for other iterables, like array? i.e.
```
import collections
if not isinstance(self._encoders, collections.Iterable):
```

Comment (jeremystan) [reply lore/pipelines/holdout.py]: 👍

**Comment by jeremystan** [comment]
@montanalow after your merge, I fixed some missing `}` and `)`, but am now getting a `RuntimeWarning: Early stopping conditioned on metric `val_loss` which is not available.` warning when running all tests that I haven't seen before. Are you?

**Comment by montanalow** [comment]
@jeremystan I've seen that before. I also just merged my fix from master w/ your fix in this branch, from a previous bad merge.

**Comment by montanalow** [comment]
@jeremystan actually this looks like a new instance coming from 
```
lore test -s tests.unit.test_models.TestNestedKeras
```
which was recently merged. I'll take a look at that.

File: lore/pipelines/holdout.py~ [hunk 2]
```diff
@@ -0,0 +1,333 @@
+from __future__ import absolute_import
```
Comment (montanalow) [inline lore/pipelines/holdout.py~]: @jeremystan Is this a tempfile created by you editor that should be removed?

Comment (jeremystan) [reply lore/pipelines/holdout.py~]: @montanalow yes, I yanked it

**Comment by jeremystan** [comment]
@montanalow looks like you approved changes, but I can't merge this into master. Are you still waiting for something?
